tests/findmmrentries/findmmrentries_test.go

```diff
@@ -0,0 +1,226 @@
+
+	// redirect std out to a known pipe so we can capture it
+	rescueStdout := os.Stdout
+	defer func() { os.Stdout = rescueStdout }() // ensure we redirect std out back after
+
+	reader, writer, _ := os.Pipe()
+	os.Stdout = writer
+
+	err := app.Run([]string{
+		"veracity",
+		"find-mmr-entries",
+		"--log-tenant", prodPublicTenant,
+	})
+	assert.NoErrorf(err, "the event is a known good event from the public production tenant, yet we have errored trying to find the mmr entries")
+
+	writer.Close()
+	actualBytes, err := io.ReadAll(reader)
+	require.NoError(err)
+
+	// convert the stdout to a string and strip the newlines
+	actual := strings.ReplaceAll(string(actualBytes), "\n", "")
+
+	assert.Equal("matches: [27899]", actual)
+}
+
+// TestAssetsV2EventAsLeafIndexStdIn tests we can find
+// the correct PROD public assetsv2 event mmr entry match as
+// a leaf index.
+func (s *FindMMREntriesSuite) TestAssetsV2EventAsLeafIndexStdIn() {
+	assert := s.Assert()
+	require := s.Require()
+
+	app := veracity.NewApp("version", true)
+	veracity.AddCommands(app, true)
+
+	// note: the suite does a before & after pipe for Stdin
+	s.StdinWriteAndClose(katdata.KnownGoodPublicAssetsV1EventLaterMassif)
+
+	// redirect std out to a known pipe so we can capture it
+	rescueStdout := os.Stdout
+	defer func() { os.Stdout = rescueStdout }() // ensure we redirect std out back after
+
+	reader, writer, _ := os.Pipe()
+	os.Stdout = writer
+
+	err := app.Run([]string{
+		"veracity",
+		"find-mmr-entries",
+		"--log-tenant", prodPublicTenant,
+		"--massif-start", "1",
+		"--massif-end", "1", // the event is in massif 1
+		"--as-leafindexes", "true",
+	})
+	assert.NoErrorf(err, "the event is a known good event from the public production tenant, yet we have errored trying to find the mmr entries")
+
+	writer.Close()
+	actualBytes, err := io.ReadAll(reader)
+	require.NoError(err)
+
+	// convert the stdout to a string and strip the newlines
+	actual := strings.ReplaceAll(string(actualBytes), "\n", "")
+
+	assert.Equal("matches: [13952]", actual)
+}
+
+// TestAssetsV2EventWrongMassifStdIn tests we CANNOT find
+// the correct PROD public assetsv2 event mmr entry match
+// if we set the range of massifs to not include the massif the event is in.
+func (s *FindMMREntriesSuite) TestAssetsV2EventWrongMassifStdIn() {
+	assert := s.Assert()
+	require := s.Require()
+
+	app := veracity.NewApp("version", true)
+	veracity.AddCommands(app, true)
+
+	// note: the suite does a before & after pipe for Stdin
+	s.StdinWriteAndClose(katdata.KnownGoodPublicAssetsV1EventLaterMassif)
+
+	// redirect std out to a known pipe so we can capture it
+	rescueStdout := os.Stdout
+	defer func() { os.Stdout = rescueStdout }() // ensure we redirect std out back after
+
+	reader, writer, _ := os.Pipe()
+	os.Stdout = writer
+
+	err := app.Run([]string{
+		"veracity",
+		"find-mmr-entries",
+		"--log-tenant", prodPublicTenant,
+		"--massif-start", "0",
```

Comment:
I like this test, But setting the range to be zero length will likely excersise the short circute do nothing path. Just take this as a minor comment, I think these tests are great. Its hard to be sure we have enough massifs to do a non zero scan that ommits the massif containing the event

Reply:
I agree, we need to ramp up the public events :P I will revist the test when we have a later massifs in the prod public tenant.

It doesn't short circuit, it considers the first massif only (index 0) and there are two massifs